crates/proto/src/error.rs

```diff
@@ -369,20 +418,22 @@ impl ProtoError {
     /// Constructor to NX type errors
     #[inline]
     pub fn nx_error(
-        query: Query,
-        soa: Option<Record<SOA>>,
+        query: Arc<Query>,
+        soa: Option<Arc<Record<SOA>>>,
```

Comment:
IMO it would be good to split this (the `Arc`-wrapping) out in a separate commit.

(Does the `Arc`-wrapping still make sense if we don't need to `clone()` as much because we can transfer ownership in `From` impls?)

Does it actually make sense to require that the caller wraps `Query` in `Arc` or do all the callers end up wrapping it right then? If the latter, better to wrap it in this function? (Maybe the same for `soa`.)

Reply:
Most of the cloning is gone now, so I backed out the Arc wrapping on existing fields (still need it on the authorities field in ForwardData,) and went back to Box<Query> and Option<Box<Record<SOA>> for the query and soa fields, to minimize breaking changes to the existing NoRecordsFound interface (removing the Box broke the resolve utility for instance.)